codec-socks/src/main/java/io/netty/handler/codec/socksx/v5/Socks5AddressDecoder.java

```diff
@@ -35,16 +35,30 @@
 
         @Override
         public String decodeAddress(Socks5AddressType addrType, ByteBuf in) throws Exception {
+            int readableBytes = in.readableBytes();
             if (addrType == Socks5AddressType.IPv4) {
+                if (readableBytes < 4) {
+                    return null;
+                }
                 return NetUtil.intToIpAddress(in.readInt());
             }
             if (addrType == Socks5AddressType.DOMAIN) {
-                final int length = in.readUnsignedByte();
+                if (readableBytes < 1) {
+                    return null;
+                }
+                final int length = in.getUnsignedByte(in.readerIndex());
```

Comment:
I wonder if this pattern is common enough to have a `peek*` family of methods on Buffer?

Reply:
I think the problem would be that these would need to return "boxed" values and so would be more costly. WDYT ?

Reply:
ah or you mean to return a value at the current index without advance the index ?

Reply:
yeah, the latter